net/netns/netns.go

```diff
@@ -66,3 +68,20 @@ type Dialer interface {
 	Dial(network, address string) (net.Conn, error)
 	DialContext(ctx context.Context, network, address string) (net.Conn, error)
 }
+
+func IsLocalhost(host string) bool {
+	name, _, err := net.SplitHostPort(host)
+	if err == nil {
+		host = name
```

Comment:
rather than give is variable different meanings at different times, maybe call the func arg `addr string` (which matches what it's named at call sites, from net.Dial), and then have a separate `host` variable inside the func that you assign to in two places, either from the split or from addr on split failure.

Reply:
Done.